ports/esp32/machine_pin.c

```diff
@@ -177,7 +177,11 @@ STATIC const machine_pin_obj_t machine_pin_obj[] = {
     {{NULL}, -1}, // 23 not a pin
     {{NULL}, -1}, // 24 not a pin
     {{NULL}, -1}, // 25 not a pin
-    {{NULL}, -1}, // 26 FLASH/PSRAM
+    #if CONFIG_ESP32_SPIRAM_SUPPORT
```

Comment:
This constant is ESP32 specific, ie won't work on ESP32S2/S3.  I think `CONFIG_SPIRAM` can be used as a generic value, but I didn't test it.

Reply:
@dpgeorge thanks and sorry!

Fixed as you proposed.

Not sure if CONFIG_SPIRAM is really generic (it isn't defined in [boards/sdkconfig.spiram](https://github.com/micropython/micropython/blob/master/ports/esp32/boards/sdkconfig.spiram)) but I checked it is defined in 
[boards/sdkconfig.spiram_sx](https://github.com/micropython/micropython/blob/master/ports/esp32/boards/sdkconfig.spiram_sx) what is enough for this particular purpose. To be sure I poisoned one of #if CONFIG_SPIRAM branches and compiled for BOARD=GENERIC_S3_SPIRAM and BOARD=GENERIC_S2